src/lib/uri/uri.h

```diff
@@ -15,6 +15,10 @@ extern "C" {
 
 struct uri_param;
 
+/**
+ * WARNING: this structure is exposed in Lua via FFI (see src/lua/uri.lua): any
+ * change  must be reflected in `ffi.cdef`.
+ */
```

Comment:
Nit: I see no point in factoring out this change into a separate patch (it touches the same structures that are patched in the main patch; it introduces no changes). Let's merge this change into the main patch.

Reply:
This change has no logical connection with the optimization patch, that's why I put it into a separate commit.

Reply:
My point is that there's no point in splitting this out - it doesn't ease code review, it doesn't ease cherry-picking, it doesn't help bisecting. You basically add comments to structures that you patch in the same commit. Splitting it out is useless.

Reply:
Merged it, though I can't say I completely agree with you.